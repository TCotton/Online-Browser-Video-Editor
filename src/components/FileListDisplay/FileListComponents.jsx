import React from 'react';
import PropTypes from 'prop-types';

export const FileListComponents = (props) => {
    const {data} = props;

    return (
        <section className="file-row" data-testid='FileListComponents'>
            {data ? data.map(x => (x.name ?
                    <section className="file-row" key={x.name}>
                        <div className="column-1">
                            <img src={`./file.svg`} alt="file" className="icon"/>
                        </div>
                        <div className="column-2">
                            <p data-testid="name">{x.name}</p>
                        </div>
                        <div className="column-3">
                            <p data-testid="size">{(x.size / 1000 / 1000).toFixed()}MB</p>
                        </div>
                    </section> : null
            )) : null}
        </section>
    )
}

FileListComponents.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string.isRequired,
        size: PropTypes.number.isRequired,
        type: PropTypes.string,
        lastModified: PropTypes.number
    })).isRequired
}