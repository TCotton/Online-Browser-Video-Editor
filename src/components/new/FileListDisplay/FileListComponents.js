import React from 'react';

export const FileListComponents = () => {
    return (
        <div data-testid='FileListComponents' />
    )
}