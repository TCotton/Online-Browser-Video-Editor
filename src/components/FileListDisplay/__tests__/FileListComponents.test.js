import React from 'react';
import {render} from '@testing-library/react';
import {FileListComponents} from "../FileListComponents";

describe('FileListComponent', () => {

    const props = {
        data: [{
            id: 1,
            name: 'string',
            size: 999,
            type: 'string',
            lastModified: 999
        }]
    }

    it('should render successfully', () => {
        const {container} = render(<FileListComponents {...props} />);
        expect(container.querySelector("[data-testid=\"FileListComponents\"]")).not.toBeNull();
    });

    it('should match snapshot', () => {
        const {baseElement} = render(<FileListComponents  {...props} />);
        expect(baseElement).toMatchSnapshot();
    });

    it('should match values',  () => {
        const {container} = render(<FileListComponents {...props} />);
        expect(container.querySelector("[data-testid=\"name\"]").textContent).toBe(props.data[0].name);
        expect(container.querySelector("[data-testid=\"size\"]").textContent).toBe('0MB');
    });
});