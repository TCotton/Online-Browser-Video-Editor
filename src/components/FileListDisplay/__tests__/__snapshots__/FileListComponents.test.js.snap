// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileListComponent should match snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="FileListComponents"
    />
  </div>
</body>
`;
